src/Lean/Data/Json/FromToJson.lean

```diff
@@ -28,6 +28,12 @@ instance : ToJson Json := ⟨id⟩
 instance : FromJson JsonNumber := ⟨Json.getNum?⟩
 instance : ToJson JsonNumber := ⟨Json.num⟩
 
+instance : FromJson Empty where
+  fromJson? j := throw (s!"type Empty has no constructor to match JSON value '{j}'. "
+  ++ "This is likely due to invalid JSON input, such as e.g. constructing an `Option Empty` "
+  ++ "with a non-null value.")
```

Comment:
I think that this isn't quite true - the problem isn't the validity of the JSON, it's the fact that it's being used to target a type that it shouldn't. And the bit about `null` leaks the encoding in a way that I think isn't so helpful.

What about this phrasing?
```suggestion
  ++ "This occurs when deserializing a value for type Empty, e.g. at type "
  ++ "Option Empty with code for the 'some' constructor.")
```

A [string gap](https://lean-lang.org/blog/2024-2-4-lean-450/) would also be nice here.

Reply:
Sure. I went back-and-forth with the wording, trying to decide if I should make the error relative to the input format or the target type.  Happy to change.